### PR TITLE
hotfix[g3.1-g3.4]: Generation of rooms and priority search

### PR DESCRIPTION
**Pull request by ogikuzma**
Here is the priority search. Generation of rooms is tested, but the priority search is not. Since there was a misunderstanding with the deadline I needed to make this hotfix so that the web team gets this in hands as quickly as possible. If there is any problem with priority search it will be resolved today, so that everything works fine for tomorrows daily.

### CODE REVIEW

**Review by jelenabudisa — DISMISSED** [review]

File: HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs
Comment (milana-todorovic) [inline HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs]: Shouldn't there be a block here? The tab makes it kind of confusing, but the if from line 79 currently only refers to line 80.

File: HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs
Comment (milana-todorovic) [inline HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs]: I feel like adding a GetDoctorsBySpecialty method to the repo would've been easier, if there isn't one already? Although, maybe it would've been bit of a nasty Linq expression, something like <code>return context.Doctors.Where(doctor => doctor.DoctorSpecialties.Any(specialty => specialty.Id == specialtyId));</code>.
Note: I am not asking you to change this, I'm aware that you need this branch joined quickly. It's just an observation for possible future improvements.

File: HealthcareSystem/Backend/Service/RoomAndEquipment/RoomService.cs
Comment (milana-todorovic) [inline HealthcareSystem/Backend/Service/RoomAndEquipment/RoomService.cs:107]: I feel like this could be implemented in a much simpler way, maybe through a nice database query with some joins, or just though using some fields (although, again, that might lead to some clunky Linq expressions). However, when I went to look for better ways to implement this, here's what I noticed - why isn't EquipmentInRooms annotated with foreign keys at all? I see that the correct key is set in OnModelCreating, but it does not have the correct foreign key constraints.
Note: I am not asking you to change this either, I'm aware that you need this branch joined quickly. However, the foreign key thing is an issue that should be resolved at some point.

File: HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs
Comment (ogikuzma) [reply HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs]: Yes, that is correct. My colleague added that 80th line and probably forgot to add a block. It will be fixed.

File: HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs
Comment (ogikuzma) [reply HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs]: Unfortunately, GetDoctorsBySpecialty method or similar cannot be found in the repository, but it would make sense to add it to the repository instead of checking the speciality in the doctors class. I believe that method from doctors class could stay there for now, since maybe it will be needed in the future, but I will add GetDoctorsBySpecialty into the repository.

File: HealthcareSystem/Backend/Service/RoomAndEquipment/RoomService.cs
Comment (ogikuzma) [reply HealthcareSystem/Backend/Service/RoomAndEquipment/RoomService.cs:107]: As database query is concerned, that is what I thought of also, but the expressions would really look nasty and a bit unreadable. I will try to think of maybe 2 separate queries that would make it shorter and clearer, but it will need to wait for now.
As foreign keys are concerned, I am not really sure why there aren't any foreign key annotations although it is obvious that the class needs them. I won't change that now in order not to break anything, but I will definitely add a to-do for that.

File: HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs
Comment (milana-todorovic) [reply HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs]: Fixed in 10f7608cb77efe6e2952e70b5775072ef31e3872

File: HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs
Comment (milana-todorovic) [reply HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs]: Applied in 10f7608cb77efe6e2952e70b5775072ef31e3872.

**Comment by milana-todorovic** [comment]
I'm not sure what happened with the pipelines here, but it's not actually failed tests, it's just some sort of failure with package fetching during execution. I'm running the jobs again to see if it resolves.

**Review by milana-todorovic — DISMISSED** [review]
Seems alright.

**Comment by ogikuzma** [comment]
@milana-todorovic I've just seen that and did builds afterward and everything seems fine.

**Comment by milana-todorovic** [comment]
@ogikuzma I checked manually and all tests pass on this branch as well. Since this is some sort of issue with either Azure DevOps or our pipeline that probably only manifested on this branch since it's the most recently updated one, I will merge this PR as soon as you get the second approval despite the pipeline failure.

**Comment by milana-todorovic** [comment]
It seems like the pipeline issue self resolved after a couple attempts at rerunning it, so everything is fine as far as that's concerned.

**Review by vlaksi — DISMISSED** [review]
Lgtm

File: HealthcareSystem/GraphicalEditorServer/Mappers/ExaminationMapper.cs
Comment (milana-todorovic) [inline HealthcareSystem/GraphicalEditorServer/Mappers/ExaminationMapper.cs]: Did this need to be deleted and was forgotten, or?

File: HealthcareSystemFrontends/GraphicalEditor/Service/AppointmentService.cs
Comment (milana-todorovic) [inline HealthcareSystemFrontends/GraphicalEditor/Service/AppointmentService.cs]: This as well.

File: HealthcareSystemFrontends/GraphicalEditor/Service/GenericHTTPService.cs
Comment (milana-todorovic) [inline HealthcareSystemFrontends/GraphicalEditor/Service/GenericHTTPService.cs:56]: Post method should probably be edited at some point to take object and handle the JSON conversion like this one does.

**Review by milana-todorovic — DISMISSED** [review]
Seems alright.

**Review by jelenabudisa — DISMISSED** [review]

File: HealthcareSystem/Backend/Model/DTO/BasicAppointmentSearchDTO.cs
Comment (milana-todorovic) [inline HealthcareSystem/Backend/Model/DTO/BasicAppointmentSearchDTO.cs]: You cannot delete this method, since the [PatientWebApp uses it](https://github.com/PSW-2020-ORG4/HealthcareSystem/pull/131/checks?check_run_id=1540395859). Please fix the build issues.

File: HealthcareSystem/GraphicalEditorServer/Mappers/ExaminationMapper.cs
Comment (ogikuzma) [reply HealthcareSystem/GraphicalEditorServer/Mappers/ExaminationMapper.cs]: This will be deleted.

File: HealthcareSystemFrontends/GraphicalEditor/Service/AppointmentService.cs
Comment (ogikuzma) [reply HealthcareSystemFrontends/GraphicalEditor/Service/AppointmentService.cs]: This will be deleted.

File: HealthcareSystemFrontends/GraphicalEditor/Service/GenericHTTPService.cs
Comment (ogikuzma) [reply HealthcareSystemFrontends/GraphicalEditor/Service/GenericHTTPService.cs:56]: Yes, that is right. An issue that still continues to be unhandled...

**Review by milana-todorovic — APPROVED** [review]

**Review by jelenabudisa — APPROVED** [review]

**Pull request merged**
